src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -68,28 +64,26 @@ internal LanguageWorkerChannel()
            string workerId,
            string rootScriptPath,
            IScriptEventManager eventManager,
-           IWorkerProcessFactory processFactory,
-           IProcessRegistry processRegistry,
            WorkerConfig workerConfig,
-           Uri serverUri,
+           ILanguageWorkerProcess languageWorkerProcess,
            ILoggerFactory loggerFactory,
            IMetricsLogger metricsLogger,
            int attemptCount,
-           ILanguageWorkerConsoleLogSource consoleLogSource,
            bool isWebHostChannel = false,
            IOptions<ManagedDependencyOptions> managedDependencyOptions = null)
         {
             _workerId = workerId;
             _rootScriptPath = rootScriptPath;
             _eventManager = eventManager;
-            _processFactory = processFactory;
-            _processRegistry = processRegistry;
             _workerConfig = workerConfig;
-            _serverUri = serverUri;
-            _workerChannelLogger = loggerFactory.CreateLogger($"Worker.{workerConfig.Language}.{_workerId}");
-            _workerCapabilities = new Capabilities(_workerChannelLogger);
-            _consoleLogSource = consoleLogSource;
+            _runtime = workerConfig.Language;
             _isWebHostChannel = isWebHostChannel;
+            _loggerFactory = loggerFactory;
```

Comment:
Is `_loggerFactory` truly needed?

Reply:
This is being used to create a worker logger with category that used id and runtime: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs#L84

Let me know if there is another way to achieve this.

Reply:
fixed this.